onedal/ensemble/forest.py

```diff
@@ -206,6 +206,9 @@ def _get_onedal_params(self, data):
                 ceil(
                     self.min_samples_split * n_samples)))
 
+        rs = check_random_state(self.random_state)
+        seed = rs.randint(0, np.iinfo("i").max)
```

Comment:
Isn't this redundant? What happens if we don't provide a seed? Naively I would expect that it picks a random one as default. What is the behavior of `check_random_state()` if something is wrong?
If it throws, fine, but if not, you may have to add some error handling here.

Reply:
This matches the daal4py code, check_random_state comes from sklearn internals which follow similar behavior. (https://github.com/intel/scikit-learn-intelex/blob/master/daal4py/sklearn/ensemble/_forest.py#L689)